### PR TITLE
Issue1421

### PR DESCRIPTION
**Pull request by Vj-L**
### Describe your changes :

I worked on this to avoid or Cancel Github Actions on a new push

#
### Type of change :

- [x] Bug fix - Issue 1421

### CODE REVIEW

**Comment by ayush-shah** [comment]
fixes #1421

File: .github/workflows/java-checkstyle.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ on:
   pull_request_target:
     branches: [ main ]
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/java-checkstyle.yml:21]: let's update the group name to be `{workflow_name}-{github_head_ref}`. For Example - `group: java-checkstyle-${{ github.head_ref }}`

File: .github/workflows/maven-build.yml [hunk 2]
```diff
@@ -23,6 +23,9 @@ on:
       - "catalog-rest-service/**"
       - "openmetadata-ui/**"
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/maven-build.yml:27]: We need to use fallback value here as there are multiple events that this CI is listening to. 
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

Comment (akash-jain-10) [inline .github/workflows/maven-build.yml:27]: let's update the group name to be {workflow_name}-{github_head_ref}.

File: .github/workflows/openmetadata-ingestion-core-version-comment.yml [hunk 3]
```diff
@@ -20,6 +20,9 @@ on:
     paths:
     - 'catalog-rest-service/src/main/resources/json/schema/**'
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/openmetadata-ingestion-core-version-comment.yml:24]: let's update the group name to be {workflow_name}-{github_head_ref}.

File: .github/workflows/py-checkstyle.yml [hunk 4]
```diff
@@ -17,6 +17,9 @@ on:
   pull_request_target:
     branches: [ main ]
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/py-checkstyle.yml:21]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

File: .github/workflows/py-generate.yml [hunk 5]
```diff
@@ -18,6 +18,9 @@ on:
     paths:
       - 'catalog-rest-service/src/main/resources/json/**'
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/py-generate.yml:22]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

File: .github/workflows/py-ingestion-core-publish.yml [hunk 6]
```diff
@@ -24,6 +24,9 @@ on:
       - 'catalog-rest-service/src/main/resources/json/schema/**'
       - 'ingestion-core/src/metadata/_version.py'
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/py-ingestion-core-publish.yml:28]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

Comment (akash-jain-10) [inline .github/workflows/py-ingestion-core-publish.yml:28]: `github.head_ref` is only defined on Pull Requests. Let's change this to a more common variable that will be available on all events and is unique

File: .github/workflows/py-tests.yml [hunk 7]
```diff
@@ -23,6 +23,9 @@ on:
       - ingestion/**
       - catalog-rest-service/**
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/py-tests.yml:27]: We need to use fallback value here as there are multiple events that this CI is listening to.
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

Comment (akash-jain-10) [inline .github/workflows/py-tests.yml:27]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

File: .github/workflows/selenium-noIngestion-tests.yml [hunk 8]
```diff
@@ -26,6 +26,9 @@ on:
       - 'openmetadata-ui/src/main/resources/ui/**'
       - 'openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pagesWithoutData/**'
 
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/selenium-noIngestion-tests.yml:30]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

Comment (akash-jain-10) [inline .github/workflows/selenium-noIngestion-tests.yml:30]: We need to use fallback value here as there are multiple events that this CI is listening to.
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

File: .github/workflows/selenium-tests.yml [hunk 9]
```diff
@@ -25,6 +25,9 @@ on:
     paths:
       - 'openmetadata-ui/src/main/resources/ui/**'
       - 'openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/**'
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/selenium-tests.yml:29]: We need to use fallback value here as there are multiple events that this CI is listening to.
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

Comment (akash-jain-10) [inline .github/workflows/selenium-tests.yml:29]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

File: .github/workflows/yarn-coverage.yml [hunk 10]
```diff
@@ -14,6 +14,9 @@ on:
       - openmetadata-ui/src/main/resources/ui/**
 env:
   UI_WORKING_DIRECTORY: openmetadata-ui/src/main/resources/ui
+concurrency: 
+  group: ${{ github.head_ref }}
```
Comment (akash-jain-10) [inline .github/workflows/yarn-coverage.yml:18]: We need to use fallback value here as there are multiple events that this CI is listening to.
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

Comment (akash-jain-10) [inline .github/workflows/yarn-coverage.yml:18]: let's update the group name to be `{workflow_name}-{github_head_ref}`.

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: .github/workflows/java-checkstyle.yml (hunk 1 above)
Comment (Vj-L) [reply .github/workflows/java-checkstyle.yml]: I have updated as per the workflow name

File: .github/workflows/openmetadata-ingestion-core-version-comment.yml (hunk 3 above)
Comment (Vj-L) [reply .github/workflows/openmetadata-ingestion-core-version-comment.yml]: I have updated as per the workflow name

File: .github/workflows/py-checkstyle.yml (hunk 4 above)
Comment (Vj-L) [reply .github/workflows/py-checkstyle.yml]: I have updated the group name as per the workflow name

**Review by Vj-L — COMMENTED** [review]
I have updated the changes as per the comments
Kindly review and let me know if there is anychange

**Review by akash-jain-10 — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**